homeassistant/components/azure_servicebus/notify.py

```diff
@@ -0,0 +1,83 @@
+"""Support for azure service bus notification."""
+import json
+import logging
+
+from azure.servicebus.aio import Message, ServiceBusClient
+import voluptuous as vol
+
+from homeassistant.components.notify import (
+    ATTR_DATA,
+    ATTR_MESSAGE,
+    ATTR_TARGET,
+    ATTR_TITLE,
+    PLATFORM_SCHEMA,
+    BaseNotificationService,
+)
+from homeassistant.const import CONTENT_TYPE_JSON
+import homeassistant.helpers.config_validation as cv
+
+CONF_CONNECTION_STRING = "connection_string"
+CONF_QUEUE_NAME = "queue"
+CONF_TOPIC_NAME = "topic"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_CONNECTION_STRING): cv.string,
+        vol.Exclusive(
+            CONF_QUEUE_NAME, "output", "Can only send to a queue or a topic."
+        ): cv.string,
+        vol.Exclusive(
+            CONF_TOPIC_NAME, "output", "Can only send to a queue or a topic."
+        ): cv.string,
+    }
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_service(hass, config, discovery_info=None):
+    """Get the notification service."""
+    connection_string = config[CONF_CONNECTION_STRING]
+    queue_name = config.get(CONF_QUEUE_NAME)
+    topic_name = config.get(CONF_TOPIC_NAME)
+
+    servicebus = ServiceBusClient.from_connection_string(
+        connection_string, loop=hass.loop
```

Comment:
Does the library support asyncio? Should we use the home assistant async api instead, ie `async_get_service`?

Reply:
The library does support asyncio, but it doesn't do any IO or async calls during setup so I picked the sync method because of that. Changed it to `async_get_service` now and tested it, and it works. I'm not sure I have completely understood everything around asyncio, so please correct me if that doesn't work.

Reply:
It's good to stay in async context as long as we can. There's some overhead to switching context.

If the library supports asyncio it's most likely intended to run within the event loop by default, even when it's not strictly necessary.

Many asyncio functions are not threadsafe and must be run within the event loop to work. Eg getting the current event loop with `asyncio.get_event_loop` works inside the event loop, in callbacks and coroutines.

Reply:
We need to change back to use `get_service` since the library is implemented very weirdly. See below.

Reply:
Changed back to sync method now, and hopefully added a good enough comment to warn about this.